sympy/solvers/inequalities.py

```diff
@@ -982,3 +985,453 @@ def reduce_inequalities(inequalities, symbols=[]):
+        if type(rel) == Ne or isinstance(rel, AppliedBinaryRelation) and rel.function == Q.ne:
+            raise TypeError("'not equal to' is not allowed in linear programming.")
+        # convert AppliedBinaryRelation to Relational
+        if isinstance(rel, AppliedBinaryRelation):
+            if rel.function == Q.le:
+                rel = Le(rel.lhs, rel.rhs, evaluate=False)
+            elif rel.function == Q.ge:
+                rel = Ge(rel.lhs, rel.rhs, evaluate=False)
+            elif rel.function == Q.eq:
+                rel = Eq(rel.lhs, rel.rhs, evaluate=False)
+        # sanity check
+        if not isinstance(rel, (Le, Ge, Eq)):
+            raise TypeError(f"Unrecognized relation: {rel}")
+        # make Relational canonical
+        rel = rel.canonical
+        # put in expression form
+        if not isinstance(rel, Eq):
+            eqns.append(rel.lts - rel.gts)
+        else:
+            # x = 3 can be represented as x <= 3 and x >= 3
+            eqns.append(rel.lhs - rel.rhs)
+            eqns.append(rel.rhs - rel.lhs)
+
+    obj_const = f.subs({var: 0 for var in variables})
+
+    A, B = linear_eq_to_matrix(eqns, *variables)
+    C = linear_eq_to_matrix(f, *variables)[0]
+    D = ImmutableMatrix([obj_const])
+
+    eqns = A*Matrix(variables)
+    standard_constraints = [eqns[i] <= B[i] for i in range(len(B))]
+
+    return A, B, C, D, standard_constraints
+
+
+def linear_optimization(f, cons, variables):
+    """
+    Finds values for variables that minimize the linear objective
+    function `f` subject to linear constraints `cons`. All coefficients must be
+    Rational or floats.
+
+    Parameters
+    ==========
+
+    f : a linear objective function to minimize
+        If you want to minimize a function, f, simply make the objective
+        function -f.
+
+    cons : list of linear inequalities
+        Strict inequalities (>, <) and not equals (!=) are not allowed.
+
+    variables : a list of all variables included in constraints and objective
+
+    Returns
+    =======
+
+    Returns `(optimum, argmax, argmin_dual)`:
+
+    optimum : Rational or Float
+
+    argmax : dictionary of variables to Rationals or Floats
+
+    argmin_dual : dictionary of constraints to Rationals or Floats
+        Contains values for dual variables that minimize dual problem.
+
+    Examples
+    ========
+
+    >>> from sympy.abc import x, y, z
+    >>> from sympy.solvers.inequalities import linear_optimization
+    >>> r1 = y+2*z <= 3
+    >>> r2 = -x-3*z <= -2
+    >>> r3 = 2*x+y+7*z <= 5
+    >>> optimum, argmax, argmin_dual  = linear_optimization(x+y+5*z, [r1,r2,r3], [x, y, z])
+    >>> optimum
+    11/3
+    >>> argmax
+    {x: 0, y: 1/3, z: 2/3}
+    >>> argmin_dual
+    {-x - 3*z <= -2: 2/3, y + 2*z <= 3: 0, 2*x + y + 7*z <= 5: 1}
+
+    See Also
+    ========
+
+    solve_univariate_inequality
+    linprog_from_matrices
+    find_feasible
```

Comment:
You removed `find_feasible`, didn't you?

Reply:
yeah. I plan to add it in a separate PR, but this should probably be removed from this PR for now.